src/components/search/index.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { useState } from 'react';
+import { useRef } from 'react';
+import { useEffect } from 'react';
+
+import Input from '../input';
+
+import './index.css';
+
+const Search = ({
+  length,
+  multi,
+  placeholder,
+  multiPlaceholder,
+  onSearch,
+  onKeySelect,
+  SingleComponent,
+  MultiComponent
+}) => {
+  const [focused, setFocused] = useState(false);
+  const [expanded, setExpanded] = useState(false);
+  const [highlightedIndex, setHighlightedIndex] = useState(-1);
+  const clearFunc = useRef(() => null);
+
+  const onFocus = () => setFocused(true);
+  const onBlur = () => setFocused(false);
+  const onChangeExpanded = (expanded) => setExpanded(expanded);
+  const getClearFunc = (func) => (clearFunc.current = func);
+
+  const onKeyDown = (event) => {
+    if (!focused || expanded || !length)
+      return;
+
+    switch (event.key) {
+      case 'ArrowUp':
+        event.preventDefault();
+        if (highlightedIndex > 0)
+          setHighlightedIndex(highlightedIndex - 1);
+        break;
+
+      case 'ArrowDown':
+        event.preventDefault();
+        if (highlightedIndex < length - 1)
+          setHighlightedIndex(highlightedIndex + 1);
+        break;
+
+      case 'Enter':
+        event.preventDefault();
+        if (highlightedIndex < 0 || highlightedIndex > length - 1)
+          break;
+        onKeySelect(highlightedIndex);
+        clearFunc.current();
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  useEffect(() => {
+    if (focused && !expanded && length)
+      setHighlightedIndex(0);
+    else
+      setHighlightedIndex(-1);
+  }, [expanded, focused, length]);
```

Comment:
Is `[expanded, focused, length]` on this line the return value of the arrow function? It seems unusual.

Reply:
No that's a react hooks thing. Any time one of those values changes, the `useEffect` will be run. They're essentially memoization inputs.

Reply:
My bad. I thought it was a function definition.